components/script/dom/webglrenderingcontext.rs

```diff
@@ -376,17 +371,17 @@ impl WebGLRenderingContext {
         // if it is UNSIGNED_SHORT_5_6_5, UNSIGNED_SHORT_4_4_4_4,
         // or UNSIGNED_SHORT_5_5_5_1, a Uint16Array must be supplied.
         // If the types do not match, an INVALID_OPERATION error is generated.
-        let received_size = if let Some(data) = data {
-            if unsafe { array_buffer_view_data_checked::<u16>(data).is_some() } {
+        let received_size = if data.is_null() {
```

Comment:
What do you mean? `data` is the pointer given by codegen, and can be null if you pass `null`.

In general, I'd be more comfortable if `object?` was represented in codegen by `Option<NonZero<*mut JSObject>>`, but...